### DISCUSSION TITLE
Raspberrymatic in a docker container, can't connect to sensor

### DISCUSSION THREAD

**Discussion by feu77** [OpenCCU/OpenCCU#1180] [🇺🇸 Q&A] [answered]
Hello,
I used to run RM on my RPi3 with a HM MOD RPI PCB which is reading HM-Sen-Wa-Od and din't have any issues. Now I switched to an RPi4 and run everything in a container incl. RM. The installation was no problem, 100% following the instructions by Jens Maus. The web UI starts, no errors. But I can't connect my HM-Sen-Wa-Od.
This is what the logs say:

```
Starting watchdog...
Identifying onboard hardware: oci, OK
Initializing RTC Clock: no hardware found
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Starting logging: OK
Populating /dev using udev: done
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.18.0.2, OK
Identifying Homematic RF-Hardware: ...HmRF: none, HmIP: none, OK
Updating Homematic RF-Hardware: no GPIO/USB connected RF-hardware found
Preparing start of hs485d: no Hm-Wired hardware found
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: no configuration file
Starting ssdpd: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: no Hm-Wired hardware found
Starting multimacd: not required
Starting rfd: no BidCos-RF hardware found
Starting HMIPServer: ....OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
```
I read it as if my HM MOD RPI PCM isn't detected, can someone confirm this?
What other logs do you want me to post?
Any idea?

BTW, I'm running on the same RPi a CC2531 USB stick and a SSD on USB which work without a problem. Can they interfere with the HM MOD RPI PCB? The CC2531 is about 50 cm away from the RPi .

Many thanks!
Chris

**Comment by feu77** [comment, marked as answer]
Update: I hot unplugged the module which caused a system crash (LOL) and I got this in the logs after reboot:

Identifying Homematic RF-Hardware: ....HmRF: HM-MOD-RPI-PCB/GPIO, HmIP: HM-MOD-RPI-PCB/GPIO, OK


Updating Homematic RF-Hardware: HM-MOD-RPI-PCB: 2.8.6, OK

And I could pair my Sensor. So everything is fine. I hope the rest still works. I checked the CC2531 and it's ok.

**Reply by jens-maus** [reply]
Well, hot unplugging something from the GPIOs is certainly not a good idea and obviously by reconnecting the module you probably used the correct pins now ;)

**Comment by feu77** [comment]
@jens-maus it was not a good idea, but I did use the right pins. It just happened now again, that on a reboot, the HM-MOD-RPI-PCB wasn't discovered once again. I used the 3,3v pin (PIN17), that's why I'm sure I used the right pins the first time.

Do you have an idea why it's not discovered everytime on startup?

BTW, I speak German, if you prefer.. ;-)

**Comment by feu77** [comment]
I used PIN17 for a fan, not for the module..
